### PR TITLE
Insight bug fixes

### PR DESCRIPTION
**Pull request by dominikl**
# What this PR does

Some bugfixes for Insight.

- If an image has been open for a while ( > 10min) in the image viewer and you try to save modified rendering settings when closing the image viewer an error dialog popped up (but the rendering settings are saved). Reported by @mporter-gre .
- If an image is open in the image viewer and the server is restarted, the rendering engine did not get reloaded, and on every change of the rendering settings (e.g. switch a channel on/off), Insight "thought" it lost connection to the server and tried to reconnect again. This was unfortunately a side effect of #5513 . Reported by @jburel on 5.4.2 release testing ( row 9 https://docs.google.com/spreadsheets/d/1iRCuIDLTA0k7QeZK1xLwSXbYdirUoNIpNXnDLBbdQLE/edit#gid=420881033 ).

I leave this PR open and gradually add some more bugfixes. As Insight is in maintenance mode this will be either critical ones or easy-to-fix ones. 

# Testing this PR

Basically try to replicate the bugs described in above section.

### CODE REVIEW

File: components/insight/SRC/org/openmicroscopy/shoola/agents/metadata/rnd/RendererModel.java [hunk 1]
```diff
@@ -960,6 +960,7 @@ void setColorModel(String colorModel)
 		if (rndControl == null) return;
 		makeHistorySnapshot();
 		rndControl.setModel(colorModel);
+		System.out.println("jep!");
```
Comment (joshmoore) [inline components/insight/SRC/org/openmicroscopy/shoola/agents/metadata/rnd/RendererModel.java]: Really? :)

Comment (dominikl) [reply components/insight/SRC/org/openmicroscopy/shoola/agents/metadata/rnd/RendererModel.java]: Oops :-)

Comment (sbesson) [reply components/insight/SRC/org/openmicroscopy/shoola/agents/metadata/rnd/RendererModel.java]: feature!

File: components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java [hunk 2]
```diff
@@ -2397,6 +2410,12 @@ private RenderingEnginePrx generateRenderingEngine(
 			service.lookupPixels(pixelsID);
 			needDefault(pixelsID, service);
 			service.load();
+            Set<Long> pixIds = renderingEngines.get(ctx);
```
Comment (jburel) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java:2413]: for big images several rendering engines are created for the same set of pixels
Have you considered that case?

Comment (dominikl) [reply components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java:2413]: Shouldn't be a problem, works like before. Only the pixels Id is needed to reload the RenderingEngines, it's only used here: https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/insight/SRC/org/openmicroscopy/shoola/env/data/DataServicesFactory.java#L365 (respectively bit further down at line 388 to reload the RenderingControl for the pixels Id (which should take care of the different RenderingEngines)).

**Comment by pwalczysko** [comment]
Note: if further fixes are to be implemented here as the header suggests, then https://trello.com/c/0c3vG5rs/49-insight-failed-login-message

**Comment by pwalczysko** [comment]
Testing of 

>f an image has been open for a while ( > 10min) in the image viewer and you try to save modified rendering settings when closing the im

all works fine here, tested on Win7 laptop and Mac, small and big images.

**Comment by pwalczysko** [comment]
Testing of

>If an image is open in the image viewer and the server is restarted, the rendering engine did not get reloaded, and on every change of the rendering settings (e.g. switch a channel on/off), Insight "thought" it lost connection

Did the tests locally on my localhost, as described in step 3 of https://docs.openmicroscopy.org/internal/testing_scenarios/ConnectionErrorHandling.html. I am a bit confused about the report because it says that the step 3 was tested, but it says "when a dataset is expanded". There is only a mention about full viewer and switching channels on and off in step 3. 

I did:

  - open a big or small image in Full Viewer (identical results obtained for both)
  - restart the server 
  - switch a channel on and off in the still opened Full Viewer
  
Result:
   
  - an intermittent "Trying to reconnect..." window popped up, but vanished after couple of seconds
  - the initial action was not executed (i.e. the channel switch-off which elicited the dialog got not executed) but every following action on the same Full Viewer was performed as expected

Another test:

   - expand a project
   - restart the server
   - expand a dataset -> result is that this just works, no popup or detrimental behaviour observed.

**Comment by pwalczysko** [comment]
In summary:
The first test https://github.com/openmicroscopy/openmicroscopy/pull/5650#issuecomment-365586928 -> all as expected.
Although some question about the precise setup of https://github.com/openmicroscopy/openmicroscopy/pull/5650#issuecomment-365594600 remain, the behaviour seems to me as fully satisfactory.

Not sure if @dominikl wants to add more fixes to this PR, (such as https://github.com/openmicroscopy/openmicroscopy/pull/5650#issuecomment-365586620), but if not, ready to merge FMPOV.

**Comment by dominikl** [comment]
Thanks @pwalczysko . The steps for testing point 2 were absolutely correct. Without the bugfix Insight would hang in the "Trying to reconnect..." loop forever. As you noticed there's still the (imho) minor problem, that the first rendering setting action on an open image after a server restart is not executed (fails silently), but that would be quite tricky to fix.

The failed login error message shouldn't be difficult to fix. Not sure if it's already too late for 5.4.4? I'll open a separate PR anyway, let you decide if it'll go in to 5.4.4 or later.

**Comment by jburel** [comment]
Connection message handled in another PR https://github.com/openmicroscopy/openmicroscopy/pull/5658
